sqlmesh/core/test/definition.py

```diff
@@ -331,6 +344,11 @@ def runTest(self) -> None:
 
             self.assert_equal(expected, actual, sort=sort, partial=partial)
 
+    def _execute(self, query: exp.Expression) -> pd.DataFrame:
+        """Executes the query with the engine adapter and returns a DataFrame."""
+        with patch.dict(self._engine_adapter_dialect.generator_class.TRANSFORMS, self._transforms):
```

Comment:
Does `Dialect.get_or_raise(self.engine_adapter.dialect)` return a global instance or a new one each time? If it's the latter, can't we just set transformers once and not patch it?

Reply:
Unfortunately we can't do this, because `TRANSFORMS` is a class attribute and since dialects are singletons we'd end up mutating every dialect if we didn't patch it.